src/client/components/pages/parts/admin-panel-search-results.tsx

```diff
@@ -0,0 +1,160 @@
+import PrivsEditModal from './privs-edit-modal';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {faPencilAlt} from '@fortawesome/free-solid-svg-icons';
+import {genEntityIconHTMLElement} from '../../../helpers/entity';
+import {getPrivilegeShieldIcon} from '../../../../common/helpers/privileges-utils';
+
+
+const {Button, Table} = bootstrap;
+
+type AdminPanelSearchResultsState = {
+	selectedUser?: Record<string, unknown>,
+	showModal: boolean
+};
+type AdminPanelSearchResultsProps = {
+	results?: any[],
+	updateResultsOnPrivsChange: () => void,
+	user: Record<string, unknown>
+};
+
+
+/**
+ * Renders the document and displays the 'AdminPanelSearchResults' page.
+ * @returns {ReactElement} a HTML document which displays the AdminPanelSearchResults.
+ * @param {object} props - Properties passed to the component.
+ */
+class AdminPanelSearchResults extends React.Component<AdminPanelSearchResultsProps, AdminPanelSearchResultsState> {
+	static displayName = 'AdminPanelSearchResults';
+
+	static propTypes = {
+		results: PropTypes.array,
+		updateResultsOnPrivsChange: PropTypes.func.isRequired,
+		user: PropTypes.object.isRequired
+	};
+
+	static defaultProps = {
+		results: null
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			selectedUser: null,
+			showModal: false
+		};
+		this.onCloseModal = this.onCloseModal.bind(this);
+	}
+
+	onCloseModal() {
+		this.setState({showModal: false});
+	}
+
+	openPrivsEditModal(user) {
+		this.setState({
+			selectedUser: user,
+			showModal: true
+		});
+	}
+
+	render() {
+		const noResults = !this.props.results || this.props.results.length === 0;
+		const results = this.props.results.map((result) => {
+			if (!result) {
+				return null;
+			}
+			const name = result.defaultAlias ? result.defaultAlias.name :
+				'(unnamed)';
+			const link = `/editor/${result.bbid}`;
+
+			/* eslint-disable react/jsx-no-bind */
+			return (
+				<tr key={result.bbid}>
+					<td>
+						<a href={link}>
+							{genEntityIconHTMLElement('editor')}
+							{name}
+						</a>
+					</td>
+					<td>
+						<PrivilegeBadges privs={result.privs}/>
+					</td>
+					<td>
+						<Button
+							variant="link"
+							onClick={() => this.openPrivsEditModal(result)}
+						>
+							<img height="18" src={getPrivilegeShieldIcon(result.privs)}/> {' '}
+							<FontAwesomeIcon icon={faPencilAlt}/>
+						</Button>
+					</td>
+				</tr>
+			);
+		});
+		const tableCssClasses = 'table table-striped';
+		if (noResults) {
+			return null;
+		}
+		return (
+			<div>
+				{
+					this.state.showModal &&
+					(
+						<div>
+							<PrivsEditModal
+								adminId={this.props.user.id}
+								handleCloseModal={this.onCloseModal}
+								show={this.state.showModal}
+								targetUser={this.state.selectedUser}
+								updateResultsOnPrivsChange={this.props.updateResultsOnPrivsChange}
+							/>
+						</div>
+					)
+				}
+				<h3 style={{color: '#754e37'}}>
```

Comment:
CSS classes instead of inlined styles, please :)
It's easier to reason about the CSS and keep it all synchronized when it is all in one place instead of scattered across the codebase.

Reply:
Indeed. I have created a class named `search-results-heading` in this [PR](https://github.com/metabrainz/bookbrainz-site/pull/998). Since this `AdminPanelSearchResults` component is largely inspired from the `SearchResults` component, I have made this change in that file as well. It was a small enough change to not warrant another PR, and so I took the liberty of including this change too.